torch/xpu/__init__.py

```diff
@@ -16,11 +17,40 @@
 from .streams import Event, Stream
 
 _initialized = False
+_tls = threading.local()
 _initialization_lock = threading.Lock()
+_queued_calls: List[
+    Tuple[Callable[[], None], List[str]]
+] = []  # don't invoke these until initialization occurs
 _is_in_bad_fork = getattr(torch._C, "_xpu_isInBadFork", lambda: False)
 _device_t = Union[_device, str, int, None]
 
 
+class _LazySeedTracker:
```

Comment:
Can we factor that out to not duplicate the impl?

Reply:
Is it reasonable to refactor these code in an additional PR, see [refactor code to share across different devices](https://github.com/pytorch/pytorch/pull/120602). When upstreaming, we would like not to affect the CUDA code.